docs/source/DeveloperGuide/AddingNewComponents/Utilities/DErrno/index.rst

```diff
@@ -33,106 +33,106 @@ There are two problems with this approach:
+          res = complicatedStuff(item)
 
-      if res['OK']:
-        successful[item] = res['Value']
-      else:
-        print "Oh, there was a problem: %s"%res['Message']
-        failed[item] = "Could not perform doSomething"
+          if res["OK"]:
+              successful[item] = res["Value"]
+          else:
+              print(f"Oh, there was a problem: {res['Message']}")
+              failed[item] = "Could not perform doSomething"
 
-    return S_OK('Successful' : successful, 'Failed : failed)
+    return S_OK("Successful" : successful, "Failed": failed)
 
 
-.. _DError:
+.. _DErrno:
 
-----------------------------------
-DError
-----------------------------------
+------
+DErrno
+------
 
-In order to address the problems raised earlier, the DError object has been created. It contains an error code, as well as a technical message. The human readable generic error message is inherent to the error code, in a similar way to what *os.strerror* is doing.
+In order to address the problems raised earlier, the DErrno object has been created. It contains an error code, as well as a technical message. The human readable generic error message is inherent to the error code, in a similar way to what *os.strerror* is doing.
 
 
 .. code-block:: python
 
-  from DIRAC.Core.Utilities import DError
+  from DIRAC.Core.Utilities import DErrno
   import errno
 
   def func1():
-    # Error happening here, with an interesting technical message
-    return DError(errno.ENOENT, 'the interesting technical message')
+      # Error happening here, with an interesting technical message
+      return DErrno(errno.ENOENT, "the interesting technical message")
 
 
 The interface of this object is fully compatible with S_ERROR
 
 .. code-block:: python
 
-  res = DError(errno.ENOENT, 'the interesting technical message')
+  res = DErrno(errno.ENOENT, "the interesting technical message")
 
-  print res
+  print(res)
   # No such file or directory ( 2 : the interesting technical message)
 
-  print res['OK']
+  print(res["OK"])
   # False
 
-  print res['Message']
+  print(res["Message"])
   # No such file or directory ( 2 : the interesting technical message)
 
 
-  # Extra info of the DError object
+  # Extra info of the DErrno object
 
-  print res.errno
+  print(res.errno)
   # 2
 
-  print res.errmsg
+  print(res.errmsg)
   # the interesting technical message
 
 
-Another very interesting feature of the DError object is that it keeps the call stack when created, and the stack is displayed in case the object is displayed using *gLogger.debug*
+Another very interesting feature of the DErrno object is that it keeps the call stack when created, and the stack is displayed in case the object is displayed using *gLogger.debug*
 
-The *Derror* object replaces S_ERROR, but should also be used in the *Failed* dictionary for bulk treatments.
+The *Derrno* object replaces S_ERROR, but should also be used in the *Failed* dictionary for bulk treatments.
```

Comment:
I may be missing something while reading the documentation, it seems that we should use `DErrno` instead of `S_ERROR`. Shouldn't we apply something like:
```
S_ERROR(DErrno.ECTMPF, 'Some message')
```

Reply:
Indeed the code have changed since long time, but not this documentation. I update it.

Reply:
well, except that this is not true, unfortunately. https://github.com/DIRACGrid/DIRAC/issues/3898